Healthy/Classes/ReusableViews/CheckboxButton.swift

```diff
@@ -0,0 +1,21 @@
+import UIKit
+
+class CheckboxButton: UIButton {
+
+    // Bool property
+    var isChecked: Bool = false {
+        didSet {
+            let image = isChecked ? UIImage.icCheckboxSelected : UIImage.icCheckboxNotSelected
+            self.setImage(image, for: UIControl.State.normal)
+        }
+    }
+
+    override func awakeFromNib() {
+        self.addTarget(self, action: #selector(buttonClicked(sender:)), for: .touchUpInside)
+        self.isChecked = false
+    }
+
+    @objc private func buttonClicked(sender: UIButton) {
+        isChecked = !isChecked
+    }
+}
```

Comment:
```suggestion
import UIKit

final class CheckboxButton: UIButton {

    // MARK: Properties

    private(set) var isChecked: Bool = false {
        didSet {
            let image = isChecked ? UIImage.icCheckboxSelected : UIImage.icCheckboxNotSelected
            self.setImage(image, for: UIControl.State.normal)
        }
    }

    // MARK: Lifecycle

    override func awakeFromNib() {
        self.addTarget(self, action: #selector(buttonClicked(sender:)), for: .touchUpInside)
        self.isChecked = false
    }

    // MARK: Actions

    @objc private func buttonClicked(sender: UIButton) {
        isChecked.toggle()
    }
}
```

Reply:
Done, thanks I'll make sure to always follow this !! 👍🏼